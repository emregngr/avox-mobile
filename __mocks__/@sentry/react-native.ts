import type { ComponentType } from 'react'

const mockedSentry = {
  init: jest.fn(),
  wrap: <T extends ComponentType<any>>(Component: T): T => Component,
  withScope: jest.fn((callback: (scope: any) => void) =>
    callback({
      setTag: jest.fn(),
      setLevel: jest.fn(),
      setUser: jest.fn(),
      setContext: jest.fn(),
    }),
  ),
  captureException: jest.fn(),
  captureMessage: jest.fn(),
  addBreadcrumb: jest.fn(),
  configureScope: jest.fn(),
  setUser: jest.fn(),
  setTag: jest.fn(),
  setContext: jest.fn(),
  setRelease: jest.fn(),
  setDist: jest.fn(),
  setEnvironment: jest.fn(),
  close: jest.fn(() => Promise.resolve(true)),
  flush: jest.fn(() => Promise.resolve(true)),
}

module.exports = mockedSentry

export type mockedSentryType = typeof mockedSentry
