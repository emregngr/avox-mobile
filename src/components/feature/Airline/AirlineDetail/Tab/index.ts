export * from './AirlineFlightTab'
export * from './CompanyTab'
export * from './FleetTab'
export * from './SafetyEnvTab'
