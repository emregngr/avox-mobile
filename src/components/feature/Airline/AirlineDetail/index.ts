export * from './AirlineHeader'
export * from './AirlineRowItem'
export * from './AirlineSectionRow'
export * from './Tab'
