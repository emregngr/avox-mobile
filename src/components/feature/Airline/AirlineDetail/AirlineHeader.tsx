import { Ionicons } from '@expo/vector-icons'
import React, { useMemo } from 'react'
import { View } from 'react-native'

import { ThemedText } from '@/components/common/ThemedText'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'
import type { Airline } from '@/types/feature/airline'

interface AirlineHeaderProps {
  airlineData: Airline
}

export const AirlineHeader = ({ airlineData }: AirlineHeaderProps) => {
  const { selectedTheme } = useThemeStore()
  const { selectedLocale } = useLocaleStore()

  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const { iataCode, icaoCode, operations } = airlineData
  const { hub, region } = operations ?? {}
  const { address, name: hubName } = hub ?? {}

  const backgroundClass = useMemo(() => `bg-${region?.toLowerCase()}`, [region])

  const homeIconColor = useMemo(() => colors?.onPrimary100, [colors])

  const localeStrings = useMemo(
    () => ({
      continent: getLocale('continent'),
      iata: getLocale('iata'),
      icao: getLocale('icao'),
    }),
    [selectedLocale],
  )

  return (
    <View className={backgroundClass}>
      <View className="p-4 gap-y-4">
        <View className="flex-row items-center bg-background-primary py-3 rounded-xl overflow-hidden">
          <View className="flex-1 items-center">
            <ThemedText
              color="text-70" lineBreakMode="tail" numberOfLines={1}
              type="body4"
            >
              {localeStrings.iata}
            </ThemedText>
            <ThemedText color="text-100" type="body1">
              {iataCode}
            </ThemedText>
          </View>

          <View className="w-px h-8 bg-background-quaternary" />

          <View className="flex-1 items-center">
            <ThemedText
              color="text-70" lineBreakMode="tail" numberOfLines={1}
              type="body4"
            >
              {localeStrings.icao}
            </ThemedText>
            <ThemedText color="text-100" type="body1">
              {icaoCode}
            </ThemedText>
          </View>

          <View className="w-px h-8 bg-background-quaternary" />

          <View className="flex-1 items-center">
            <ThemedText
              color="text-70" lineBreakMode="tail" numberOfLines={1}
              type="body4"
            >
              {localeStrings.continent}
            </ThemedText>
            <ThemedText color="text-100" type="body1">
              {region}
            </ThemedText>
          </View>
        </View>

        <View className="flex-row items-center px-5 py-3 rounded-xl overflow-hidden bg-background-primary">
          <Ionicons color={homeIconColor} name="home" size={20} />
          <View className="mx-4">
            <ThemedText
              color="text-100" ellipsizeMode="tail" numberOfLines={1}
              type="body1"
            >
              {hubName}
            </ThemedText>
            <ThemedText
              color="text-70" ellipsizeMode="tail" numberOfLines={1}
              type="body2"
            >
              {address}
            </ThemedText>
          </View>
        </View>
      </View>
    </View>
  )
}
