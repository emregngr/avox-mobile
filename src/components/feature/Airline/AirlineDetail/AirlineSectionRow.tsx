import type { ReactNode } from 'react'
import React, { useMemo } from 'react'
import { View } from 'react-native'

import { ThemedText } from '@/components/common/ThemedText'

interface AirlineSectionRowProps {
  children: ReactNode
  title: string
}

export const AirlineSectionRow = ({ children, title }: AirlineSectionRowProps) => {
  const memoizedChildren = useMemo(() => children, [children])

  return (
    <View className="p-4 rounded-xl overflow-hidden bg-background-secondary">
      <ThemedText className="mb-4" color="text-100" type="h3">
        {title}
      </ThemedText>
      {memoizedChildren}
    </View>
  )
}
