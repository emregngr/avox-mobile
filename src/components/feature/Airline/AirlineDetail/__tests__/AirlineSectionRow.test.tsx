import { render } from '@testing-library/react-native'
import React from 'react'
import { Text, View } from 'react-native'

import { AirlineSectionRow } from '@/components/feature/Airline/AirlineDetail/AirlineSectionRow'

jest.mock('@/components/common/ThemedText', () => {
  const { Text } = require('react-native')

  return {
    ThemedText: ({ children, ...props }: { children: string }) => (
      <Text {...props}>{children}</Text>
    ),
  }
})

describe('AirlineSectionRow Component', () => {
  it('should render the title correctly', () => {
    const title = 'Fleet Information'
    const { getByText } = render(
      <AirlineSectionRow title={title}>
        <View />
      </AirlineSectionRow>,
    )

    expect(getByText(title)).toBeTruthy()
  })

  it('should render its children correctly', () => {
    const title = 'Contact Details'
    const childText = 'Phone: 123-456-7890'

    const { getByText } = render(
      <AirlineSectionRow title={title}>
        <Text>{childText}</Text>
      </AirlineSectionRow>,
    )

    expect(getByText(childText)).toBeTruthy()
  })

  it('should render multiple children', () => {
    const title = 'Notes'
    const firstChildText = 'This is the first note.'
    const secondChildText = 'This is the second note.'

    const { getByText } = render(
      <AirlineSectionRow title={title}>
        <Text>{firstChildText}</Text>
        <Text>{secondChildText}</Text>
      </AirlineSectionRow>,
    )

    expect(getByText(firstChildText)).toBeTruthy()
    expect(getByText(secondChildText)).toBeTruthy()
  })
})

describe('AirlineSectionRow Component Snapshot', () => {
  it('should render the AirlineSectionRow Component successfully', () => {
    const { toJSON } = render(
      <AirlineSectionRow title="Snapshot Title">
        <Text>Snapshot child content</Text>
      </AirlineSectionRow>,
    )

    expect(toJSON()).toMatchSnapshot()
  })
})
