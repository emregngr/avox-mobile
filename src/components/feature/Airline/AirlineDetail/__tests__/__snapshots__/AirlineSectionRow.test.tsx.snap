// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`AirlineSectionRow Component Snapshot should render the AirlineSectionRow Component successfully 1`] = `
<View
  className="p-4 rounded-xl overflow-hidden bg-background-secondary"
>
  <Text
    className="mb-4"
    color="text-100"
    type="h3"
  >
    Snapshot Title
  </Text>
  <Text>
    Snapshot child content
  </Text>
</View>
`;
